### PR TITLE
IDEX-3986: Implement local storage to store user data in browser

### PR DESCRIPTION
**Pull request by tolusha**

### CODE REVIEW

**Comment by vparfonov** [comment]
ok

File: commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/storage/BrowserLocalStorageImpl.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+public class BrowserLocalStorageImpl implements LocalStorage {
+
+    private final Storage storage;
+
+    public BrowserLocalStorageImpl(@Nonnull Storage storage) {
+        this.storage = storage;
+    }
+
+    /**
+     * {@inheritDoc}
```
Comment (azatsarynnyy) [inline commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/storage/BrowserLocalStorageImpl.java]: What purpose of inheritDoc tag in empty javadoc?

File: commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/storage/DummyLocalStorageProviderImpl.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.util.storage;
+
+/**
+ * Dummy {@link LocalStorageProvider}.
```
Comment (skabashnyuk) [inline commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/storage/DummyLocalStorageProviderImpl.java]: What is the purpose of this implementation?
Can you describe it in javadoc?

File: commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/storage/LocalStorage.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+package org.eclipse.che.ide.util.storage;
+
+import org.eclipse.che.commons.annotation.Nullable;
+
+import javax.annotation.Nonnull;
+
+
+/**
+ * Stores client-side date into a storage.
+ *
```
Comment (skabashnyuk) [inline commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/storage/LocalStorage.java]: date?
What is the purpose of  this interface?

File: commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/storage/BrowserLocalStorageImpl.java (hunk 1 above)
Comment (tolusha) [inline commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/storage/BrowserLocalStorageImpl.java]: I would say it is just a habit.

Comment (tolusha) [inline commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/storage/BrowserLocalStorageImpl.java]: I'll remove them. They are unnecessary now.

File: commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/storage/LocalStorage.java (hunk 3 above)
Comment (tolusha) [inline commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/storage/LocalStorage.java]: It was a typo.
It is about client-side data.

File: commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/storage/DummyLocalStorageProviderImpl.java (hunk 2 above)
Comment (tolusha) [inline commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/storage/DummyLocalStorageProviderImpl.java]: Added

**Comment by skabashnyuk** [comment]
ок

**Pull request merged**